From: Len Brown <len.brown@intel.com>
Subject: intel_idle: do not use the LAPIC timer for ATOM C2
References: fate#311818,fate#311942
Git-commit: 97e3bcac24f55fa6c3224f1bc39d08db73495cfb


Signed-off-by: Thomas Renninger <trenn@suse.de>

upstream: c25d29952b2a8c9aaf00e081c9162a0e383030cd

If we use the LAPIC timer during ATOM C2 on
some nvidia chisets, the system stalls.

https://bugzilla.kernel.org/show_bug.cgi?id=21032

Signed-off-by: Len Brown <len.brown@intel.com>

 drivers/idle/intel_idle.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/idle/intel_idle.c b/drivers/idle/intel_idle.c
index 1f53bb0..8b211cb 100644
--- a/drivers/idle/intel_idle.c
+++ b/drivers/idle/intel_idle.c
@@ -295,7 +295,7 @@ static int intel_idle_probe(void)
 
 	case 0x1C:	/* 28 - Atom Processor */
 	case 0x26:	/* 38 - Lincroft Atom Processor */
-		lapic_timer_reliable_states = (1 << 2) | (1 << 1); /* C2, C1 */
+		lapic_timer_reliable_states = (1 << 1); /* C1 */
 		cpuidle_state_table = atom_cstates;
 		break;
 
